Created guide on how to set the DA footprint

ZakAyesh opened the pull request:
**Description**
Created a guide on how to set the DA footprint

ZakAyesh committed:
feat:setting da footprint guide created

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+| ---------- | ---- | -------- | ------------ | ------------ |
+| `daFootprintGasScalar` | Number | 400  | 400 | unitless|
+
+The <strong>Data Availibility (DA) Footprint</strong> feature was introduced in the <strong>Jovian</strong> hardfork to limit the total amount of estimated compressed transaction data that can fit into a block.
```
```suggestion
The <strong>Data Availability (DA) Footprint</strong> feature was introduced in the <strong>Jovian</strong> hardfork to limit the total amount of estimated compressed transaction data that can fit into a block.
```

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+| `daFootprintGasScalar` | Number | 400  | 400 | unitless|
+
+The <strong>Data Availibility (DA) Footprint</strong> feature was introduced in the <strong>Jovian</strong> hardfork to limit the total amount of estimated compressed transaction data that can fit into a block.
+When an OP Stack chain recieves more calldata heavy transactions than can fit into the L1s availble blob space, [the Batcher](https://docs.optimism.io/chain-operators/tutorials/create-l2-rollup/op-batcher-setup#understanding-the-batchers-role)
```
```suggestion
When an OP Stack chain receives more calldata heavy transactions than can fit into the L1s available blob space, [the Batcher's](https://docs.optimism.io/chain-operators/tutorials/create-l2-rollup/op-batcher-setup#understanding-the-batchers-role)
```

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+
+The <strong>Data Availibility (DA) Footprint</strong> feature was introduced in the <strong>Jovian</strong> hardfork to limit the total amount of estimated compressed transaction data that can fit into a block.
+When an OP Stack chain recieves more calldata heavy transactions than can fit into the L1s availble blob space, [the Batcher](https://docs.optimism.io/chain-operators/tutorials/create-l2-rollup/op-batcher-setup#understanding-the-batchers-role)
+[can throttle it's capacity](https://docs.optimism.io/chain-operators/guides/configuration/batcher#batcher-sequencer-throttling). 
```
```suggestion
default behavior will [throttle it's capacity](https://docs.optimism.io/chain-operators/guides/configuration/batcher#batcher-sequencer-throttling). 
```

geoknee replied:
How about "will throttle the sequencer, causing it to build artificially small blocks".

ZakAyesh replied:
I prefer @geoknee 's suggestion here, as if we put "default behavior" it could make people think throttling is replaced by this feature, but it can still occur even with the DA Footprint (just less likely depending on the setting for both)

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+However, continuous batcher throttling may cause the base fee to drop to the [minimum base fee](https://docs.optimism.io/chain-operators/guides/features/setting-min-base-fee),
+causing unnecessary losses for the chain operator and negative user experiences such as priority fee auctions. Limiting the amount of calldata taken up by transactions in an block using the DA Footprint can reduce the need for batcher throttling and the issues that come from throttling.
+
+For every transaction, other than [deposit transactions](https://docs.optimism.io/reference/glossary#deposited-transaction), 
```
maybe we streamline this to, "For every L2 transaction processed by the sequencer, a resource..." and omit the deposit tx part

geoknee replied:
You could also say "for every L2 transaction which needs to be submitted by the batcher" (this already excludes deposits and sounds a bit more obvious IMO).

ZakAyesh replied:
The thing here is that I think that many average users still won't think about the deposit transaction flow too much, as it is unique to L2's so it's always nice to explicity call it out to train users to take them into account. 

I don't feel too strongly about this though, and do prefer the streamlined phrasing for it's concisness but its something to consider

sbvegan replied:
that is a good point, I'll leave it up to you to decide how you want to phrase it

geoknee replied:
Also don't feel strongly here.

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+For every transaction, other than [deposit transactions](https://docs.optimism.io/reference/glossary#deposited-transaction), 
+processed by an OP Stack chain, a resource called DA Footprint is tracked alongside the transaction's gas usage.
+The DA Footprint can be configured via the `daFootprintGasScalar` variable in the `SystemConfig` contract on the L1 chain.
+The Da Footprint is automatically calculated for every transaction first by calculating a `daUsageEstimate` for that transaction:
```
```suggestion
The DA Footprint is automatically calculated for every transaction first by calculating a `daUsageEstimate` for that transaction:
```

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+  - Setting the `daFootprintGasScalar` too low may prove ineffective at preventing batcher throttling or protecting against spam.
+  - When a chain's gas limit is changed, the DA footprint scales proportionally by design. However you want to retain the same absolute DA footprint limit, then you must the also change the `daFootprintGasScalar`.
+  - If no limit is set, or the value 0 is set for the daFootprintGasScalar in the SystemConfig, the default value of 400 is used
+  - If the feature is diabled, set the scalar to a very low value such as 1.
```
```suggestion
  - If the feature is disabled, set the scalar to a very low value such as 1.
```

geoknee replied:
We should actually say "in order to disable the feature, set the scalar to a very low value such as 1".

ZakAyesh replied:
done

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+
+<Warning>
+  - Setting the `daFootprintGasScalar` too high may exclude too many transactions from your blocks.
+  - Setting the `daFootprintGasScalar` too low may prove ineffective at preventing batcher throttling or protecting against spam.
```
```suggestion
  - Setting the `daFootprintGasScalar` too low may prove ineffective at preventing batcher throttling or protecting against continuous DA heavy transactions.
```

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+
+  <Step title="Call the setDAFootprintGasScalar() method">
+   <Info>
+    Note that the owner of the `SystemConfig` contract is the [System Config Owner address](https://docs.optimism.io/chain-operators/reference/privileged-roles#system-config-owner), so this transaction must be sent from that address.
```
```suggestion
    The [SystemConfig owner](https://docs.optimism.io/chain-operators/reference/privileged-roles#system-config-owner) is the only address that can make these changes.
```

sbvegan commented on chain-operators/guides/features/setting-da-footprint.mdx:
````diff
@@ -0,0 +1,100 @@
+    Example (using [cast](https://getfoundry.sh/cast/reference/cast/)):
+
+    ```
+    cast send <SYSTEM_CONFIG_PROXY_ADDRESS> "daFootprintGasScalar(uint16)" 405 
````
idk if there's a way we can fix this, but when it renders the button's cover up the test

ZakAyesh replied:
done

sbvegan reviewed (commented):
It's looking good, I've added some suggestions.

geoknee commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+---
+title: Set the DA Footprint
```
```suggestion
title: Set the DA Footprint Gas Scalar
```

geoknee commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+
+| Parameter | Type | Default | Recommended | Units |
+| ---------- | ---- | -------- | ------------ | ------------ |
+| `daFootprintGasScalar` | Number | 400  | 400 | unitless|
```
The unit is actually "gas per byte".

ZakAyesh replied:
fixed

geoknee commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+| ---------- | ---- | -------- | ------------ | ------------ |
+| `daFootprintGasScalar` | Number | 400  | 400 | unitless|
+
+The <strong>Data Availibility (DA) Footprint</strong> feature was introduced in the <strong>Jovian</strong> hardfork to limit the total amount of estimated compressed transaction data that can fit into a block.
```
```suggestion
The <strong>Data Availibility (DA) Footprint</strong> feature was introduced in the <strong>Jovian</strong> hardfork to limit the total amount of transaction data that can fit into a block based on a scaled estimate of the compressed size (or "data availability footprint") of that data.
```

ZakAyesh replied:
done

geoknee commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+    You can find its proxy address in the [state.json generated by op-deployer](https://docs.optimism.io/chain-operators/tutorials/create-l2-rollup/op-deployer-setup#deploy-l1-contracts).
+  </Step>
+
+  <Step title="Call the setDAFootprintGasScalar() method">
```
```suggestion
  <Step title="Call the `setDAFootprintGasScalar()` method">
```

Not entirely sure how this will render...

ZakAyesh replied:
@geoknee I checked and the back ticks show up in the title so I'm going to revert this change

geoknee commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+<Warning>
+  - Setting the `daFootprintGasScalar` too high may exclude too many transactions from your blocks.
+  - Setting the `daFootprintGasScalar` too low may prove ineffective at preventing batcher throttling or protecting against spam.
+  - When a chain's gas limit is changed, the DA footprint scales proportionally by design. However you want to retain the same absolute DA footprint limit, then you must the also change the `daFootprintGasScalar`.
```
Nit: I would put these last in the list of things in the warning.

ZakAyesh replied:
done

geoknee reviewed (approved):
LGTM, left some suggestions.

geoknee commented on chain-operators/guides/features/setting-da-footprint.mdx:
```diff
@@ -0,0 +1,100 @@
+
+## Overview
+
+| Parameter | Type | Default | Recommended | Units |
```
There is potential for confusion here because of the way we translate the _on chain_ default (`0`) to the off chain default (`400`). It's the off chain one that matters most. Perhaps we can solve that by having two columns?

ZakAyesh replied:
I opted to put 400 in parentheses after the zero and added a warning callout below the table explaining. Let me know if you think that solves the confusion adequately.

geoknee replied:
That works for me. Could we put a link to the specs somewhere (perhaps in the callout), just to cover the precise detail of how the default fallback works?

https://specs.optimism.io/protocol/jovian/l1-attributes.html

ZakAyesh replied:
added!

ZakAyesh committed:
Apply suggestions from code review

Co-authored-by: George Knee <georgeknee@googlemail.com>
Co-authored-by: soyboy <85043086+sbvegan@users.noreply.github.com>

ZakAyesh committed:
Apply suggestions from code review

Co-authored-by: soyboy <85043086+sbvegan@users.noreply.github.com>

ZakAyesh committed:
Apply suggestions from code review

Co-authored-by: George Knee <georgeknee@googlemail.com>
Co-authored-by: soyboy <85043086+sbvegan@users.noreply.github.com>

ZakAyesh committed:
fix:cleanup from review

ZakAyesh committed:
fix:cleanup based on review

ZakAyesh committed:
fix:clean up sentence wording

ZakAyesh committed:
cleanup

sebastianst committed:
Improve DA Footprint documentation (#1864)

Updated formatting and clarified descriptions for the DA Footprint feature and its parameters.

ZakAyesh committed:
add hyperlink to specs for default value behavior

ZakAyesh committed:
update code formatting
